### PR TITLE
Move indexed-traversable instance

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: comonad.cabal [hunk 1]
```diff
@@ -68,6 +68,17 @@ flag distributive
   default: True
   manual: True
 
+flag indexed-traversable
```
Comment (arybczak) [inline comonad.cabal:71]: Does this flag make sense considering that `indexed-traversable` has only GHC deps?

Comment (phadej) [reply comonad.cabal:71]: ditto `distributive` and `tagged` on recent enough GHCs. IMHO yes, for consistency. The implementation of this flag is quite cheap.

**Comment by cartazio** [comment]
Might it be worth just moving ci to GitHub actions? I’ll have a standardized template in a few days

**Comment by phadej** [comment]
@RyanGlScott if this is fine, then I'll go forward with making `indexed-traversable` releases, and amending `comonad` and `free` PR to not use http-tarballs.

File: comonad.cabal [hunk 2]
```diff
@@ -68,6 +68,17 @@ flag distributive
...
+  description:
+    You can disable the use of the `indexed-traversable` package using `-f-indexed-traversable`.
+    .
+    Disabling this is an unsupported configuration, but it may be useful for accelerating builds in sandboxes for expert users.
+    .
+    If disabled we will not supply instances of `indexed-traversable`
```
Comment (RyanGlScott) [inline comonad.cabal:77]: Since the description for the `-fdistributive` flag mentions the `Distributive` class, I think the description for `-findexed-traversable` should be analogous:

```suggestion
    If disabled we will not supply instances of `FunctorWithIndex`
```

File: CHANGELOG.markdown [hunk 3]
```diff
@@ -1,3 +1,8 @@
+5.0.7 [2020.xx.yy]
+------------------
+* Move `FunctorWithIndex (TracedT m w)` instance from `lens`.
+  The new `indexed-traversable` can be disabled using the flag of the same name.
```
Comment (RyanGlScott) [inline CHANGELOG.markdown:4]: This sentence reads a bit strangely to me. Perhaps rephrase it to this?

```suggestion
  This instance depends on the `indexed-traversable` package. This can be disabled using the flag of the same name.
```

**Comment by RyanGlScott** [comment]
> if this is fine, then I'll go forward with making `indexed-traversable` releases, and amending `comonad` and `free` PR to not use http-tarballs.

Sounds good to me. Thanks, @phadej!

**Comment by phadej** [comment]
This build turned green.

**Comment by RyanGlScott** [comment]
Thanks! I've uploaded [`comonad-5.0.7`](http://hackage.haskell.org/package/comonad-5.0.7) to Hackage with these changes.
